src/search_count_report.py

```diff
@@ -0,0 +1,46 @@
+import csv
+from datetime import datetime
+
+from src.report import Report
+import src.utils as utils
+
+class SearchCountReport(Report):
+
+    def generate(self):
+        # Write fields to search_count.csv
+        self.writer.writerow(['day', 'hour', 'count'])
+
+        time_to_count_dict = {}
+
+        # Read routes.csv
+        for index, row in enumerate(self.reader):
+            if index == 0:
+                continue
+            timestamp = row[0]
+            datetime = self.parse_date(timestamp)
+            hour = datetime.hour
+            day = self.parse_day(timestamp)
+            if day in time_to_count_dict:
+                if hour in time_to_count_dict[day]:
+                    time_to_count_dict[day][hour] += 1
+                else:
+                    time_to_count_dict[day][hour] = 1
+            else:
+                time_to_count_dict[day] = {hour: 1}
```

Comment:
i feel like there's a better way to write this, this `if else` pair is pretty clunky to read